citrination_client/views/client.py

```diff
@@ -12,3 +14,139 @@ def __init__(self, api_key, webserver_host="https://citrination.com", suppress_w
 
     def create(self, ml_template, search_template):
         pass
+
+    def get_available_columns(self, dataset_ids):
+        """
+        Retrieves the set of columns from the combination of dataset ids given
+
+        :param dataset_ids: The id of the dataset to retrieve columns from
+        :type dataset_ids: list of int
+        :return: A list of column names from the dataset ids given.
+        :rtype: list of str
+        """
+
+        data = {
+            "dataset_ids":
+                dataset_ids
+        }
+
+        failure_message = "Failed to get available columns in dataset(s) {}".format(dataset_ids)
+
+        return self._get_success_json(self._post_json(
+            'datasets/get-available-columns', data, failure_message=failure_message))['data']
+
+    def generate_search_template(self, dataset_ids):
+        """
+        Generates a default search templates from the available columns in the dataset ids given.
+
+        :param dataset_ids: The id of the dataset to retrieve files from
+        :type dataset_ids: list of int
+        :return: A search template based on the columns in the datasets given
+        """
+
+        data = {
+            "dataset_ids":
+                dataset_ids
+        }
+
+        failure_message = "Failed to generate a search template from columns in dataset(s) {}".format(dataset_ids)
+
+        return self._get_success_json(self._post_json(
+            'search_templates/from-dataset-ids', data, failure_message=failure_message))['data']
+
+    def prune_search_template(self, extract_as_keys, search_template):
+        """
+        Returns a new search template, but the new template has only the extract_as_keys given.
+
+        :param extract_as_keys: List of extract as keys to keep
+        :param search_template: The search template to prune
+        :return: New search template with pruned columns
+        """
+
+        data = {
+            "extract_as_keys":
+                extract_as_keys,
+            "search_template":
+                search_template
+        }
+
+        failure_message = "Failed to prune a search template"
+
+        return self._get_success_json(self._post_json(
+            'search_templates/prune-to-extract-as', data, failure_message=failure_message))['data']
+
+    def create_machine_learning_template(self, search_template, ml_config):
```

Comment:
Let's not do this here. I'd rather separate concerns - I created `ModelTemplateClient` and `SimplifiedModelTemplateBuilder` stubs for this purpose

Reply:
I'd move this into `SimplifiedModelTemplateBuilder`